fuzzers/aflcc_no_orig_dic/fuzzer.py

```diff
@@ -0,0 +1,378 @@
+        compiler='/afl/aflc-clang-fast++',
+        flags=cppflags,
+        target=fuzz_target,
+        ldflags=ldflags)
+    if os.system(bin2_cmd) != 0:
+        raise ValueError('command "{command}" failed'.format(command=bin2_cmd))
+
+    # The no-collision split-condition optimized dictionary.
+    print('[post_build] Generating no-collision-all-opt build')
+    os.environ['AFL_COVERAGE_TYPE'] = 'NO_COLLISION'
+    os.environ['AFL_CONVERT_COMPARISON_TYPE'] = 'ALL'
+    os.environ['AFL_DICT_TYPE'] = 'OPTIMIZED'
+    bin3_cmd = '{compiler} {flags} {target}.bc -o ' \
+    '{target}-no-collision-all-opt {ldflags}'.format(
+        compiler='/afl/aflc-clang-fast++',
+        flags=cppflags,
+        target=fuzz_target,
+        ldflags=ldflags)
+    if os.system(bin3_cmd) != 0:
+        raise ValueError('command "{command}" failed'.format(command=bin3_cmd))
+
+    print('[post_build] Copying afl-fuzz to $OUT directory')
+    # Copy out the afl-fuzz binary as a build artifact.
+    shutil.copy('/afl/afl-fuzz', os.environ['OUT'])
+
+
+def build():
+    """Build benchmark."""
+    prepare_build_environment()
+
+    utils.build_benchmark()
+
+    print('[post_build] Extracting .bc file')
+    fuzz_targets = get_fuzz_targets()
+    for fuzz_target in fuzz_targets:
+        post_build(fuzz_target)
+
+
+def run_fuzzer(input_corpus,
+               output_corpus,
+               target_binary,
+               additional_flags=None,
+               hide_output=False):
+    """Run afl-fuzz."""
+    # Spawn the afl fuzzing process.
+    # FIXME: Currently AFL will exit if it encounters a crashing input in seed
+    # corpus (usually timeouts). Add a way to skip/delete such inputs and
+    # re-run AFL.
+    print('[run_fuzzer] Running target with afl-fuzz')
+    command = [
+        './afl-fuzz',
+        '-i',
+        input_corpus,
+        '-o',
+        output_corpus,
+        # Use no memory limit as ASAN doesn't play nicely with one.
+        '-m',
+        'none'
+    ]
+    if additional_flags:
+        command.extend(additional_flags)
+    dictionary_path = utils.get_dictionary_path(target_binary)
+    if dictionary_path:
+        command.extend(['-x', dictionary_path])
+    command += [
+        '--',
+        target_binary,
+        # Pass INT_MAX to afl the maximize the number of persistent loops it
+        # performs.
+        '2147483647'
+    ]
+    print('[run_fuzzer] Running command: ' + ' '.join(command))
+    output_stream = subprocess.DEVNULL if hide_output else None
+    subprocess.check_call(command, stdout=output_stream, stderr=output_stream)
+
+
+def fuzz(input_corpus, output_corpus, target_binary):
+    """Run fuzzer."""
+    prepare_fuzz_environment(input_corpus)
+
+    # Note: dictionary automatically added by run_fuzzer().
+
+    # Use a dictionary for original afl as well.
+    print('[fuzz] Running AFL for original binary')
+    # src_file = '{target}-normalized-none-nopt.dict'.format(target=target_binary)
```

Comment:
All these commented source should be removed.

Reply:
Done.